### PR TITLE
Don't redefine builtin str

### PR DESCRIPTION
**Pull request by yvaucher**
**Description of the issue/feature this PR addresses:**

Dangerous redefinition of a builtin (thus targeting master)

**Current behavior before PR:**

Same

**Desired behavior after PR is merged:**

Same

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yvaucher** [comment]
@qdp-odoo FYI

**Comment by qdp-odoo** [comment]
indeed :+1: 

@Gorash can you please add this in your ongoing refactoring of the reconciliation widget?

**Comment by chermed** [comment]
@qdp-odoo  @grindtildeath  what about https://github.com/odoo/odoo/pull/22392#discussion-diff-163271600L720 ?

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -713,21 +713,21 @@ def get_reconciliation_proposition(self, account_id, partner_id=False):
         return []
 
     @api.model
-    def domain_move_lines_for_reconciliation(self, str):
+    def domain_move_lines_for_reconciliation(self, search_str):
         """ Returns the domain from the str search
             :param str: search string
         """
         if not str:
```
[the remark this answers is not in the document]
Comment (yvaucher) [reply addons/account/models/account_move.py]: good catch

**Comment by yvaucher** [comment]
Fixed missing change (exactly the kind we want to avoid) plus a docstring that I missed.

Rebased on last master.

**Comment by Gorash** [comment]
This changes is in the https://github.com/odoo/odoo/pull/22274 refactoring

**Pull request closed without merging**